pkg/p2p/libp2p/libp2p.go

```diff
@@ -390,6 +404,7 @@ func (s *Service) NewStream(ctx context.Context, overlay swarm.Address, headers
 
 	// exchange headers
 	if err := sendHeaders(ctx, headers, stream); err != nil {
+		_ = stream.Reset()
```

Comment:
why do you need to reset it?

Reply:
So, I have checked the libp2p examples, and it seem that they do `stream.Reset()` when they want to close the other side of the stream as well, if something fails for example. And they do `stream.Close()` if everything was ok and they want to close just this side of the stream.